hoomd/md/HarmonicDihedralForceCompute.cc

```diff
@@ -246,11 +252,13 @@ void HarmonicDihedralForceCompute::computeForces(unsigned int timestep)
 
 /////////////////////////
 // FROM LAMMPS: sin_shift is always 0... so dropping all sin_shift terms!!!!
+// Adding charmm dihedral functionality, sin_shift not always 0 
 /////////////////////////
 
         Scalar sign = m_sign[dihedral_type];
-        p = p*sign;
-        dfab = dfab*sign;
+        Scalar phi_0 = m_phi_0[dihedral_type];
+        p = p*sign + dfab*sin(phi_0);
```

Comment:
You should use `fast::sin` here, as in the GPU code, and also store the value of the `sin` to reuse explicitly.